invenio_rdm_records/services/communities/components.py

```diff
@@ -0,0 +1,60 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 CERN.
+#
+# Invenio-RDM-Records is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""Record communities service components."""
+
+from invenio_communities.communities.records.systemfields.access import (
+    CommunityAccess,
+    VisibilityEnum,
+)
+from invenio_communities.communities.services.components import (
+    CommunityAccessComponent as BaseAccessComponent,
+)
+from invenio_communities.communities.services.components import (
+    CustomFieldsComponent,
+    FeaturedCommunityComponent,
+    OAISetComponent,
+    OwnershipComponent,
+    PIDComponent,
+)
+from invenio_records_resources.services.records.components import (
+    MetadataComponent,
+    RelationsComponent,
+)
+
+from ...proxies import current_rdm_records_service
+from ..errors import InvalidCommunityVisibility
+
+
+class CommunityAccessComponent(BaseAccessComponent):
+    """Community access component."""
+
+    def update(self, identity, data=None, record=None, **kwargs):
+        """Update handler."""
+        self._populate_access_and_validate(identity, data, record, **kwargs)
+        if VisibilityEnum(record.access.visibility) == VisibilityEnum.RESTRICTED:
+            assert CommunityAccess.validate_visibility_level("public")
+
+            total = current_rdm_records_service.search(
+                identity, q=f"access.record:public"
+            ).total
+            if total > 0:
+                raise InvalidCommunityVisibility(
+                    reason="Cannot restrict a community with public records."
+                )
+
+
+CommunityServiceComponents = [
```

Comment:
Comment: Here we could import `DefaultCommunityComponents` instead of importing each individual cmp to save some lines of code. However, since there is already a task to improve this in a nicer way it's good enough for me.